### PR TITLE
[dtensor][TP] check funcol calls and improve doc for loss parallel

### PR DESCRIPTION
**Pull request by tianyu-l**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #121366

Since CommDebugMode is fixed, we can check that loss parallel is working as expected.

Under loss parallel, the forward computation should invoke 3 all-reduces, and the backward computation should invoke no functional collectives.

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @wconstab @yf225 @chauhang

### CODE REVIEW

File: torch/distributed/tensor/parallel/loss.py [hunk 1]
```diff
@@ -26,10 +26,13 @@
...
-    computation can be performed when the input is sharded on the class dimension. Within this context
-    manager, one can use :func:`~torch.nn.functional.cross_entropy` or :class:`~torch.nn.CrossEntropyLoss`
-    as usual, with the following assumptions on the input parameters.
+    A context manager that enables loss parallelism, where efficient parallelized loss computation
+    can be performed when the input is sharded on the class dimension. Currently only the cross-entropy
+    loss is supported.
+
+    Within this context manager, one can use :func:`~torch.nn.functional.cross_entropy` or
+    :class:`~torch.nn.CrossEntropyLoss` as usual, with the following assumptions on the input parameters.
+    The corresponding backward call, if any, also needs to happen under this context manager.
```
Comment (wanchaol) [inline torch/distributed/tensor/parallel/loss.py:35]:
```suggestion
    The corresponding ``backward()`` call, if any, also needs to happen under this context manager.
```

**Review by wanchaol — APPROVED** [review]
lgtm!
